### ISSUE TITLE
[TB] other Typography atoms don't reflect Font Settings atom changes

### ISSUE THREAD

**Issue by aaronreed708** [finos/a11y-theme-builder#311]
## Problem/Concern

It looks like if you change the default fonts for your design system in ThemeBuilder in the Typography->Font Settings atom, the other Typography atoms don't pick up the change and still show "Open Sans" as their selected value.

__To recreate__:

1. I loaded the Google Fonts "Roboto" and "Merriweather" onto my macbook using these instructions: https://flourishonline.com/download-google-fonts-mac-pc/
2. I load the pre-defined "Sample" design system into Theme Builder
3. I click on Typography->Font Settings from the left nav bar
4. I change the primary front to "Roboto" and the secondary font to "Merriweather"
5. I click on Typography->Body Styles from the left nav bar

The entry for "Body 1" still says that the font is Open Sans.  However, the description in the Font Settings page says, "The primary font is used for the body and small font styles."  If I go to edit the font family for that entry, the new primary and secondary names are not in the list of selectable values.

We will also need to decide:

- what font family values should be in the dropdown in the Edit modal dialogs.  Only primary and secondary or also any previously selected values for primary and secondary?
- what should the behavior be when new fonts are selected in  Font Settings?  Should everything just switch?  Or should they keep their currently selected value and the user has to go in and manually change each one?  Or do all of the values get overwritten?

## Proposed Solution

The behavior in the system should match what is described in the text.

**Comment by brycecurtis** [comment]
The font family should use either --primaryFont or --secondaryFont instead of being hard coded by the SDK.  The user is selecting either the var(--primaryFont) or var(--secondaryFont) font variable for all of the typography properties.  The select dropdown should show "Primary Font: Open Sans", "Secondary Font: Roboto" so they know which font name is currently assigned to primary and secondary font variables.

**Comment by evangk6** [comment]
My changes are available and ready:
https://github.com/discoverfinancial/a11y-theme-builder/pull/322
![image](https://user-images.githubusercontent.com/36457549/234655745-17ae22c3-059b-4333-9867-c0ff5d765e95.png)

PLEASE NOTE my comment in the PR stating that SDK changes are needed:

> This requires changes to the SDK. The SDK needs default values for all typography styles. This amounts to primaryFont for everything, except header/displays which use secondaryFont. Currently the SDK uses Open Sans for each, no matter what the Primary or Secondary Font is.

After these changes are made, we can merge both, and get someone to verify.

**Comment by smithbk** [comment]
I've pushed a fix to my init branch and talked and Evan is using & verifying it before it is merged.

The fontFamily.getSelectableValues method now dynamically builds the selectables of the format that Bryce suggested.
I also added a getFont() method which returns the actual font that should be used based on the currently selected primary or secondary value.  For example, if you go to the fontsSettings atoms and change the secondary font and then call displayAndHeaderStyles.displayStyles[0].getFont(), it will return the new secondary font value set on the fontSettings atom.  This assumes that the default value for the displayAndHeaderStyles.displayStyles[0].fontFamily has not been changed from "Secondary Font: xxx".  If it was changed to "Primary Font: xxx", then it will of course return the value of the fontsSettings primary font.
